core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableRelFactories.java

```diff
@@ -68,7 +70,9 @@
       implements org.apache.calcite.rel.core.RelFactories.ProjectFactory {
     @Override public RelNode createProject(RelNode input, List<RelHint> hints,
         List<? extends RexNode> childExprs,
-        @Nullable List<? extends @Nullable String> fieldNames) {
+        @Nullable List<? extends @Nullable String> fieldNames,
+        Set<CorrelationId> variablesSet) {
+      Preconditions.checkArgument(variablesSet.isEmpty());
```

Comment:
Would you please add a clarification message?

Reply:
done